test/attr/attr_availability.swift

```diff
@@ -590,6 +590,17 @@ func testFactoryMethods() {
   Int.factory2(other: 1) // expected-error {{'factory2(other:)' has been replaced by 'Int.init(other:)'}} {{3-15=Int}}
 }
 
+struct DeprecatedInit {
+  @available(*, deprecated, renamed: "init(new:)")
+  init(old: Int) {}
+
+  init(new: Int) {}
+
+  init(test: Int) {
+    self.init(old: test) // expected-warning {{'init(old:)' is deprecated: replaced by 'init(new:)'}} expected-note {{use 'init(new:)' instead}} {{15-18=new}}
+  }
+}
+
```

Comment:
Should we add a test case for `super` as well? 
```swift
class A {
  @available(*, deprecated, renamed: "init(new:)")
  init(old: Int) {}
  
  init(new: Int) {}
  
  init(test: Int) {
  }
}
class B: A {
  override init(test: Int) {
     super.init(old: test) // expected-warning {{'init(old:)' is deprecated: replaced by 'init(new:)'}} expected-note {{use 'init(new:)' instead}} {{15-18=new}}
   }
}
```
Not sure this `isa<TypeExpr>` logic going to catch this ...

Reply:
Good idea, I added one! 

I also noticed the following: obviously, when subclasses inherit initializers from their superclass, the availability attribute on that initializer is inherited too. But it turned out that the message and rename information is _not_ inherited by the generated availability attribute. So I fixed that in a second commit.